### DISCUSSION TITLE
How to run models with multi products and/or how to get volume contributions by var

### DISCUSSION THREAD

**Discussion by allisonwiener** [google/lightweight_mmm#254] [Q&A]
Hi all, I have two questions...

1. Does Lightweight MMM support multi-product models? I want to run a couple of different SKUs from one brand in one model. How should I go about that? (This was asked in Jan but remains unanswered.)

3. Where/how can I get volume contributions from each variable (media and all other variables) at a weekly level? I need to build waterfall charts for latest vs prev 52 wks. 

Thank you very much!

**Comment by DiddyDaw** [comment]
In general, when building multi-product models in MMM, you would construct seperate models for each product. I'm assuming that the products you mention are substituable for each other. For example, if you're looking at a 100ml and 250ml Shampoo of the same type, you would either model total sales (sum the volume) or build two models making sure you include cross-product variables - in particular price and distribution.

If the different SKUs are something like "flavours" - e.g. Mint, Lemon then you should be fine to model them at an aggregated level.

Afraid i don't yet know LightweightMMM well enough to give the precise code and i'm having issues running it but in the docs (https://lightweight-mmm.readthedocs.io/en/latest/api.html), the command "lightweight_mmm.plot.create_media_baseline_contribution_df(media_mix_model: [lightweight_mmm.lightweight_mmm.LightweightMMM](https://lightweight-mmm.readthedocs.io/en/latest/api.html#lightweight_mmm.lightweight_mmm.LightweightMMM), target_scaler: Optional[[lightweight_mmm.preprocessing.CustomScaler](https://lightweight-mmm.readthedocs.io/en/latest/api.html#lightweight_mmm.preprocessing.CustomScaler)] = None, channel_names: Optional[Sequence[str]] = None)→ pandas.core.frame.DataFrame[[source]](https://lightweight-mmm.readthedocs.io/en/latest/_modules/lightweight_mmm/plot.html#create_media_baseline_contribution_df)[](https://lightweight-mmm.readthedocs.io/en/latest/api.html#lightweight_mmm.plot.create_media_baseline_contribution_df)" seems to be what you want.

**Reply by allisonwiener** [reply]
Q1: Thanks for the response! Unfortunately it’s a minimum requirement to run multiple SKUS (1 brand with multiple SKUs below) in 1 model. Is it not possible to do in lightweight_mmm? Specifically I need the ability to break out volume contributions, coeffs, etc for individual SKUs & overall brand. 

Q2: Thanks! I will look further into this.

**Reply by DiddyDaw** [reply]
With respect to Q1, I think by far your best bet is to run the analysis multiple times for different SKUs. It shouldn't be too problematic to aggregate models. However if you can't do that, you could try to use the Geo Level (regional) approach but changing the data from regions to SKUs to reflect each item as the cross-section rather than the traditional regional identifier. A big warning on this - it is potentially very misleading since you won't be able to easily include effects that occur between the SKUs (substitution effects).

**Reply by allisonwiener** [reply]
Any thoughts on this? I found an article where they are running different geos & SKUs (brands/channel combos) in one model where each column is a brand/channel. For example there are 2 brands and 14 ad groups so they put 28 variables in the model and then layer in multi geos on top of that. This seems to be a multi SKU / multi geo approach. Are there any downsides to running the model in this way? 

https://forecastegy.com/posts/how-to-create-a-marketing-mix-model-with-lightweightmmm/
